scalalib/src/mill/scalalib/bsp/ScalaMetalsSupport.scala

```diff
@@ -3,23 +3,37 @@ package mill.scalalib.bsp
 import mill.api.experimental
 import mill.{Agg, T}
 import mill.define.Target
+import mill.scalalib.api.Util.isScala3
 import mill.scalalib.{Dep, DepSyntax, ScalaModule}
 
 /*+ Enable some common settings required to properly support Metals Language Server (via BSP). */
 @experimental
 trait ScalaMetalsSupport extends ScalaModule {
 
   /** The semanticDB version to use. It needs to support your configured Scala versions. */
-  def semanticDbVersion: T[String]
+  def semanticDbVersion: T[String] = T { "4.4.30" }
```

Comment:
So I'm a little unsure what to do here. Making the user fill this in makes sense _only_ for Scala 2 and I'm afraid that making it mandatory when doing nothing for Scala 3 will be confusing. Since most people will look at Scalameta to get the latest published version for Scala 3 only to realize it's not published since the compiler produces it natively in Scala 3.

sbt for example gives it a default, but the only negative part about this is that it needs to be available for the version of Scala you're using. So if 2.13.8 comes out tomorrow, someone quickly changes their version, then indeed they'll need to set it to the newest until mill is released with a newer default.

What are your thoughts on this?

Reply:
I try to keep out default versions as much as possible. Beside the cons you already mentioned, they typically don't age well and result in future maintenance burden. Eventually Mill is stable enough so that it's not necessary to upgrade, but that means versions will stay old, in most cases in disadvantage of the user. Also, once a setting has a default, it's hard to remove it without breakage. My rule of thumb is to avoid default versions.

As this target isn't useful for Scala 3, we either make it optional by type and provide a default `None`, or we could provide an empty String and fail with an explicit error in case it was not overridden but we are on Scala 2.

Reply:
Alright I went the `""` route.